datasource/src/main/scala/quasar/physical/s3/impl/children.scala

```diff
@@ -144,17 +144,17 @@ object children {
     val listingQuery = (bucket / "")
 
     val listType = ("list-type", "2").some
+    val delimiter = ("delimiter", "%2F").some
     // Converts a pathy Path to an S3 object prefix.
     val objectPrefix = aPathToObjectPrefix(dir)
     val prefix = objectPrefix.map(("prefix", _))
 
     val ct0 = ct.map(_.value).map(("continuation-token", _))
 
-    val queryUri = List(listType, prefix, ct0).unite.foldLeft(listingQuery) {
-      case (uri0, (param, value)) => uri0.withQueryParam(param, value)
-    }
+    val q = Query.fromString(s3EncodeQueryParams(
```

Comment:
Not for this PR, but I wonder if we can handle the encoding within `RequestSigning`.

Reply:
Yeah, not sure how we can do this, but it would be nice if we can simplify things in this area

Reply:
I'm going to open a ticket to keep track of this.